### DISCUSSION TITLE
Python libraries

### DISCUSSION THREAD

**Discussion by vvitzilaios** [gstavrinos/ai_msc_robotic_systems_assignments#5] [Q&A] [answered]
I have cloned the template inside src folder of my ros workspace as we discussed in the end of our yesterday class. When I open a python file (e.g. a launch file), the imports of python libraries are undefined. For example trying to import anything from launch results to an error. I used anaconda3 for interpreter as well as python native as an alternative. I also tried to install dependencies (using both conda and pip) but I get an error with some versioning conflicts. Is this a bad placement of the template directory or python things? I cannot provide a picture or a log of the error as I am attending Mr. Vouros' lesson right now 🙄

**Comment by gstavrinos** [comment, marked as answer]
Firstly, I would not recommend using python environment while using ROS. This is not the "default" way, and you might continuously encountered problems due to that.

Regardless of that, if Python is not able to find your ROS libraries, it is probably due to a faulty installation. Make sure that you have `source`'d all the required files, as specified in our lectures.

For your convenience, this line should be included in your `~/.bashrc` file, but you could also try running it inside your Python environment, before testing your script:
`source /opt/ros/humble/setup.bash`

To complete the assignments, you will also have to `source` your workspace.
